src/bin/train.cc

```diff
@@ -406,12 +408,25 @@ int main(int argc, char * argv[]) {
         config.get<string>("Corpus.test_target"),
         *src_vocab, *trg_vocab, test_max_length, test_max_length_ratio, 1);
     logger->info("Loaded 'test' corpus.");
+    const auto fmt_corpus_size = boost::format(
+        "Cleaned corpus size: train=%d dev=%d test=%d")
+        % train_sampler.getNumSamples() % dev_sampler.getNumSamples() 
+        % test_sampler.getNumSamples();
+    logger->info(fmt_corpus_size.str());
     nmtkit::BatchConverter batch_converter(*src_vocab, *trg_vocab);
 
     dynet::Model model;
 
     // Creates a new trainer.
     boost::scoped_ptr<dynet::Trainer> trainer(::createTrainer(config, &model));
+    // Disable sparse updates.
```

Comment:
Recommend to add header to clarify responsibility of comments, like:
// NOTE(MorinoseiMorizo): blah blah blah ...